app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt

```diff
@@ -180,58 +68,22 @@ suspend fun Conversation.getUnreadMessageCount(): Long? = suspendCancellableCoro
     })
 }
 
-suspend fun ConversationsClient.getConversation(sidOrUniqueName: String): Conversation = suspendCoroutine { continuation ->
-    getConversation(sidOrUniqueName, object : CallbackListener<Conversation> {
-
-        override fun onSuccess(result: Conversation) = continuation.resume(result)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.getLastMessages(count: Int): List<Message> = suspendCoroutine { continuation ->
-    getLastMessages(count, object : CallbackListener<List<Message>> {
-
-        override fun onSuccess(result: List<Message>) = continuation.resume(result)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.getMessagesBefore(index: Long, count: Int): List<Message> = suspendCoroutine { continuation ->
-    getMessagesBefore(index, count, object : CallbackListener<List<Message>> {
-
-        override fun onSuccess(result: List<Message>) = continuation.resume(result)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.sendMessage(message: Message.Options): Message = suspendCoroutine { continuation ->
-    sendMessage(message, object : CallbackListener<Message> {
-
-        override fun onSuccess(message: Message) = continuation.resume(message)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.advanceLastReadMessageIndex(index: Long): Long = suspendCoroutine { continuation ->
-    advanceLastReadMessageIndex(index, object : CallbackListener<Long> {
-
-        override fun onSuccess(index: Long) = continuation.resume(index)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
+suspend fun Conversation.sendTextMessage(attributes: Attributes, body: String?): Message {
+    return prepareMessage()
+        .setAttributes(attributes)
+        .setBody(body)
+        .buildAndSend()
 }
 
-suspend fun Conversation.getMessageByIndex(index: Long): Message = suspendCoroutine { continuation ->
-    getMessageByIndex(index, object : CallbackListener<Message> {
-
-        override fun onSuccess(message: Message) = continuation.resume(message)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
+suspend fun Conversation.sendMediaMessage(attributes: Attributes,
```

Comment:
why do we need it? what's wrong with `Conversation.sendMessage`?

Reply:
I've added these methods to be able to mock them. 
If we leave common `Conversation.sendMessage()` (just for mocking) then it would need to take `MessageBuilder` or `UnsentMessage` parameter. Wouldn't it look ugly to you?